### DISCUSSION TITLE
Use the meta watch api to refactor some mod

### DISCUSSION THREAD

**Discussion by zhang2014** [databendlabs/databend#7705] [Ideas]
Meta client supported the watch api, which makes it possible to subscribe to a key (instead of polling).

- [ ] Cluster
- [ ] Settings
- [ ] UserApi
- [ ] UserPrivilege

**Comment by drmingdrmer** [comment]
@drmingdrmer embedded meta-store needs to support watch API

**Reply by sundy-li** [reply]
> @drmingdrmer embedded meta-store needs to support watch API

Is there a faked watch API for embedded meta ?

**Reply by drmingdrmer** [reply]
Embedded meta is a burden because two sets of API have to be maintained.
Now it's a good time to discuss getting rid of embedded meta.

**Comment by sundy-li** [comment]
We need to start this task now. Make settings support watch API based on meta-service.

**Reply by drmingdrmer** [reply]
I'll be getting on this when meta-service finished upgrading from openraft 0.7 to 0.8
